### DISCUSSION TITLE
cmd_duration persistent (or not?)

### DISCUSSION THREAD

**Discussion by janos-r** [starship/starship#2052] [Q&A] [answered]
In fish shell, if you sleep 2 and enter twice, are you supposed to see the "took 2s" once or twice? Of course you get two lines, but is it intended to show the "took" statement on all consequent lines if they have no other input?

I can see a case for both possibilities, but I am using the same starship version on two different pcs and on one I get the "took" only on the next line, but on the other it persists on repeated enters.

just once (I like that better): linux mint 19.3, fish shell from the default repo
persistent: linux mint 20.0 fish shell from the default repo
![image](https://user-images.githubusercontent.com/30606201/103468226-25680680-4d57-11eb-9188-36a8f1482bd0.png)

I wonder, if it is not intended to persist... I also use it with tmux, if that could be related... I just wonder.
Thank you for any response.

**Comment by matchai** [comment, marked as answer]
We chose to have Starship not change the default behavior of how Fish Shell tracks command duration, outputted by `$CMD_DURATION`. This was done in case you have other fish tools or plugins that depend on the environment variable behaving as expected.

**Comment by janos-r** [comment]
Thank you for the answer. So keeping it is really as intended. I agree that leaving the variable to the shell is probably a good decision. On the first pc I also run byobu, that is probably influencin that env.
Once more thank you for the answer and thanks for developing this cool tool :)
